core/src/dird/reload.cc

```diff
@@ -0,0 +1,316 @@
+/*
+   BAREOS® - Backup Archiving REcovery Open Sourced
+
+   Copyright (C) 2022-2022 Bareos GmbH & Co. KG
+
+   This program is Free Software; you can redistribute it and/or
+   modify it under the terms of version three of the GNU Affero General Public
+   License as published by the Free Software Foundation and included
+   in the file LICENSE.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+   Affero General Public License for more details.
+
+   You should have received a copy of the GNU Affero General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+   02110-1301, USA.
+*/
+
+#include "dird/reload.h"
+
+#include <cassert>
+
+namespace directordaemon {
+
+/*
+ * See if two storage definitions point to the same Storage Daemon.
+ *
+ * We compare:
+ *  - address
+ *  - SDport
+ *  - password
+ */
+static inline bool IsSameStorageDaemon(StorageResource* store1,
```

Comment:
This function also exists in migrate.cc. Can it be extracted?

Reply:
yes it can